x-pack/plugins/cases/server/common/models/case_with_comments.ts

```diff
@@ -331,6 +338,38 @@ export class CaseCommentModel {
     await this.params.services.alertsService.updateAlertsStatus(alertsToUpdate);
   }
 
+  private async updateAlertsSchemaWithCaseInfo(alertAttachments: CommentRequestAlertType[]) {
+    try {
+      const alerts = getAlertInfoFromComments(alertAttachments);
+      const alertsGroupedByIndex = new Map<string, Set<string>>();
+
+      for (const alert of alerts) {
+        const idsSet = alertsGroupedByIndex.get(alert.index) ?? new Set();
+        idsSet.add(alert.id);
+        alertsGroupedByIndex.set(alert.index, idsSet);
+      }
+
+      await pMap(
```

Comment:
I talk about this in a later comment on the alerts client but my suggestion is if it doesn't cause any breaking changes within the alerts client is to modify its API to access a bulk request such that we don't need to use `pMap` here. That saves us from having to do multiple network requests to elasticsearch. And it encapsulates the complexity within the alerts client.

Taking it even further we could potentially move the group logic into the alerts client too, since I feel like it's more of the alerts client's job to ensure that it doesn't accidentally overwrite an alert in the wrong index, maybe we could just pass in an array of id and index pairs?